internal/tekton-bundle/tekton-bundle.go

```diff
@@ -38,138 +36,109 @@ type Bundle struct {
-			}
-			if err != nil {
-				return nil, err
+func decodeObjects(path string, bundle *Bundle) error {
+	file, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	decoder := yaml.NewYAMLOrJSONDecoder(file, 1024)
+	for {
+		var obj map[string]interface{}
+		err := decoder.Decode(&obj)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		if kind, ok := obj["kind"].(string); ok {
+			if kind == "" {
+				continue
 			}
-			if kind, ok := obj["kind"].(string); ok {
-				if kind == "" {
-					continue
-				}
 
-				switch kind {
-				case tasksString:
-					task := pipeline.Task{}
-					err = getObject(obj, &task)
-					if err != nil {
-						return nil, err
-					}
-					bundle.Tasks = append(bundle.Tasks, task)
-				case pipelineKindString:
-					p := pipeline.Pipeline{}
-					err = getObject(obj, &p)
-					if err != nil {
-						return nil, err
-					}
-					bundle.Pipelines = append(bundle.Pipelines, p)
-				case serviceAccountKind:
-					sa := v1.ServiceAccount{}
-					err = getObject(obj, &sa)
-					if err != nil {
-						return nil, err
-					}
-					bundle.ServiceAccounts = append(bundle.ServiceAccounts, sa)
-				case roleBindingKind:
-					rb := rbac.RoleBinding{}
-					err = getObject(obj, &rb)
-					if err != nil {
-						return nil, err
-					}
-					bundle.RoleBindings = append(bundle.RoleBindings, rb)
-				case clusterRoleKind:
-					cr := rbac.ClusterRole{}
-					err = getObject(obj, &cr)
-					if err != nil {
-						return nil, err
-					}
-					bundle.ClusterRoles = append(bundle.ClusterRoles, cr)
-				case configMapKind:
-					cm := v1.ConfigMap{}
-					err = getObject(obj, &cm)
-					if err != nil {
-						return nil, err
-					}
-					bundle.ConfigMaps = append(bundle.ConfigMaps, cm)
-				default:
-					continue
+			switch kind {
+			case tasksString:
+				task := pipeline.Task{}
+				err = getObject(obj, &task)
+				if err != nil {
+					return err
+				}
+				bundle.Tasks = append(bundle.Tasks, task)
+			case pipelineKindString:
+				p := pipeline.Pipeline{}
+				err = getObject(obj, &p)
+				if err != nil {
+					return err
+				}
+				bundle.Pipelines = append(bundle.Pipelines, p)
+			case serviceAccountKind:
+				sa := v1.ServiceAccount{}
+				err = getObject(obj, &sa)
+				if err != nil {
+					return err
 				}
+				bundle.ServiceAccounts = append(bundle.ServiceAccounts, sa)
+			case roleBindingKind:
+				rb := rbac.RoleBinding{}
+				err = getObject(obj, &rb)
+				if err != nil {
+					return err
+				}
+				bundle.RoleBindings = append(bundle.RoleBindings, rb)
+			case clusterRoleKind:
+				cr := rbac.ClusterRole{}
+				err = getObject(obj, &cr)
+				if err != nil {
+					return err
+				}
+				bundle.ClusterRoles = append(bundle.ClusterRoles, cr)
+			case configMapKind:
+				cm := v1.ConfigMap{}
+				err = getObject(obj, &cm)
+				if err != nil {
+					return err
+				}
+				bundle.ConfigMaps = append(bundle.ConfigMaps, cm)
+			default:
+				continue
 			}
```

Comment:
Can we move the error checking to the end of the switch statement?  Something like:

```go
switch kind {
   case tasksString:
     task := pipeline.Task{}
     err = getObject(obj, &task)
     bundle.Tasks = append(bundle.Tasks, task)
[...]
}
if err != nil {
  return err
}
```

Reply:
Thanks, I'll look on it. I'd love to merge this PR firstly because it's open too much time already and too many CI issues there.